providers/providers.go

```diff
@@ -27,18 +29,17 @@ func AddProvider(name string, provider Provider) {
 }
 
 // Search builds a search URL and opens it in your browser.
-func Search(p string, q string, verbose bool) {
+func Search(p string, q string, verbose bool) error {
 	builder := Providers[p]
 
 	if builder != nil {
 		url := builder.BuildURI(q)
 		if verbose {
-			fmt.Printf("%s\n", url)
+			fmt.Fprintf(os.Stderr, "%s\n", url)
 		}
-		launcher.OpenURI(url)
-	} else {
-		fmt.Printf("Provider %q not supported!\n", p)
+		return launcher.OpenURI(url)
```

Comment:
While this reduces some boilerplate, we no longer wait for the command to finish. Not important for the "open" case, but eventually if we fork a process like w3m it won't work right. I want a way to be able to pass a binary to block on, so we can support terminal browsers.